packer/builder/azure/arm/step_delete_resource_group.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See the LICENSE file in the project root for license information.
+
+package arm
+
+import (
+	"fmt"
+
+	"github.com/Azure/packer-azure/packer/builder/azure/common/constants"
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/packer"
+)
+
+type StepDeleteResourceGroup struct {
+	client *AzureClient
+	delete func(resourceGroupName string) error
+	say    func(message string)
+	error  func(e error)
+}
+
+func NewStepDeleteResourceGroup(client *AzureClient, ui packer.Ui) *StepDeleteResourceGroup {
+	var step = &StepDeleteResourceGroup{
+		client: client,
+		say:    func(message string) { ui.Say(message) },
+		error:  func(e error) { ui.Error(e.Error()) },
+	}
+
+	step.delete = step.deleteResourceGroup
+	return step
+}
+
+func (s *StepDeleteResourceGroup) deleteResourceGroup(resourceGroupName string) error {
+	res, err := s.client.GroupsClient.Delete(resourceGroupName)
+	if err != nil {
+		return err
+	}
+
+	s.client.GroupsClient.PollAsNeeded(res.Response)
+	return nil
```

Comment:
isn't there a possibility that deleting ultimately fails? don't we want to surface this?